### PR TITLE
Update runtime and sudoku to 44

### PR DESCRIPTION
**Pull request by bbhtt**
Closes https://github.com/flathub/org.gnome.Sudoku/issues/10

### CODE REVIEW

File: org.gnome.Sudoku.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "app-id": "org.gnome.Sudoku",
     "runtime": "org.gnome.Platform",
-    "runtime-version": "42",
+    "runtime-version": "43",
```
Comment (mcatanzaro) [inline org.gnome.Sudoku.json:4]: Did you try using the 44 runtime?

File: org.gnome.Sudoku.json [hunk 2]
```diff
@@ -25,8 +25,8 @@
             "buildsystem": "meson",
             "sources": [{
                 "type": "archive",
-                "url": "https://download.gnome.org/sources/gnome-sudoku/42/gnome-sudoku-42.0.tar.xz",
-                "sha256": "1d2eb4ddb8026b443645cf3585b8df1244e3828ee1c07518052b2599e1c5c28f"
+                "url": "https://download.gnome.org/sources/gnome-sudoku/44/gnome-sudoku-44.0.tar.xz",
+                "sha256": "6518d9233a46d7e13805ac787667ba4701948d9ed41a47b88797fcdba43b8fba"
```
Comment (mcatanzaro) [inline org.gnome.Sudoku.json:29]: Let's add external data checker support here so it doesn't keep getting out of date again. I just copied this from org.gnome.Chess which seems to work:

```
                "x-checker-data": {
                    "type": "json",
                    "url": "https://download.gnome.org/sources/gnome-sudoku/cache.json",
                    "version-query": ".[2][\"gnome-sudoku\"] | map(select(. | match(\"([0-9].[0-9]*[02468].[0-9]+)|([0-9]+.[0-9]+)\"; \"g\"))) | last",
                    "url-query": "\"https://download.gnome.org/sources/gnome-sudoku/\" + .[1][\"gnome-sudoku\"][$version][\"tar.xz\"]"
                }
```

**Review by mcatanzaro — APPROVED** [review]

File: org.gnome.Sudoku.json (hunk 2 above)
Comment (bilelmoussaoui) [reply org.gnome.Sudoku.json:29]: We already have a specific gnome checker..  Please use that one instead

Comment (mcatanzaro) [reply org.gnome.Sudoku.json:29]: OK, I found an example:

```
                    "x-checker-data": {
                        "type": "gnome",
                        "name": "gnome-sudoku"
                    }
```

Comment (mcatanzaro) [reply org.gnome.Sudoku.json:29]: Oh and @bilelmoussaoui could you give me access to this repo so I can merge the change please? Thanks.

File: org.gnome.Sudoku.json (hunk 1 above)
Comment (bbhtt) [reply org.gnome.Sudoku.json:4]: Oh yea, I meant to put 44. Thanks

**Comment by mcatanzaro** [comment]
Looks good now.

**Pull request merged**